docs/coding-guidelines.md

```diff
@@ -95,6 +95,14 @@ To keep naming patterns consistent across the project, enumeration patterns are
   - `pcommon.ValueTypeStr` for `pcommon.ValueType`
   - `pmetric.MetricTypeGauge` for `pmetric.MetricType`
 
+## Optional fields
+
+When representing optional fields in the configuration, we distinguish two cases:
+1. For scalars (e.g. `int`, `string`, `bool`, `time.Duration`...), the zero value of the base type
+   is used to represent the absence of the field. The default value for this field may be different
+   from the zero value, so, when using the Go API you MUST follow the guidance on the "Default
+   Configuration" section below.
```

Comment:
```suggestion
When representing optional fields in the configuration for which the zero value indicates the absence of the field, we distinguish two cases:
1. For scalars (e.g. `int`, `string`, `bool`, `time.Duration`...), the zero value of the base type
   should be used to represent the absence of the field instead of a pointer. The default value for this field may be different
   from the zero value, so, when using the Go API you MUST follow the guidance on the "Default
   Configuration" section below.
```
I think we should limit this to only applying to the case where the zero value is not a valid value for the field and can indicate absence of the field. For cases where this doesn't apply, I think we still need to evaluate whether to use pointers or an optional type.